docs/developing_iree/profiling_with_tracy.md

```diff
@@ -0,0 +1,149 @@
+TRACY_NO_EXIT=1 /data/local/tmp/iree-benchmark-module \
+  --driver=dylib \
+  --function_inputs='1x384xi32,1x384xi32,1x384xi32' \
+  --module_file=/data/local/tmp/android_module.fbvm \
+  --entry_function=serving_default
+```
+
+## Running the Tracy profiler UI, connecting and visualizing
+
+While the profile program is still running (possibly thanks to `TRACY_NO_EXIT=1`), start the Tracy profiler UI. If above you had checked out Tracy and built it in `$HOME/tracy`, then you would run:
+
+```shell
+$HOME/tracy/profiler/build/unix/Tracy-release
+```
+
+It should show a dialog offering to connect to a client i.e. a profiled program.
+
+If connecting doesn't work:
+* If the profiled program is on a separate machine, make sure you've correctly set up port forwarding.
+  * On Android, the `adb forward` many need to be run again.
+* Make sure that the profiled program is still running. Do you need `TRACY_NO_EXIT=1`?
+* Kill the profiled program and restart it.
+
+You should then start seeing a profile.
+
+Before going further, check that your profile has sampling data recorded, not just instrumentation data. Click the 'Statistics' button at the top. The window that it opens should show a 'Sampling' radio button next to 'Instrumentation'. Back to the main view, look for the 'Main thread' label on the far left. Next to it should be a small ghost icon. If either of these things are missing, you need to fix that first, because many of Tracy's interesting features rely on smapling. Most likely this is a permissions issue: make sure that you have performed the steps outlined in the above section on permissions. As an experiment, retry with the profiled program running as root. Look for any interesting `stderr` message (in the profiled program's terminal). Try `strace`'ing it.
+
+Besides sampling, let's check that you also have "systrace" working. In the initial main view, on the far left, you should see a list of CPU cores, typically labelled `CPU0`, `CPU1`, etc. To the right, a timeline view should show system-wide activity on all these CPUs - so for instance you can see context switches and you can see if anything else happened on your device that could have interfered with the benchmark's performance. If this "systrace" functionality isn't working, then again that's likely a permissions issue, perform the same troubleshooting as above about sampling.
+
+Finally, look for the part of the timeline that is of interest to you. Your area of interest might not the on the Main thread. In fact, it might be on a thread that's not visible in the initial view at all. Indeed, the initial view tends to be zoomed-in a lot. Either use the mouse directly to zoom out and navigate, to look for the 'Frame' control at the top of the Tracy window. Use the 'next frame' arrow button until more interesting threads appear. Typically, IREE generated code tends to run on a thread named `cpu0`, which is actually a thread name and unrelated to `CPU0` from the systrace view.
```

Comment:
"might not **be** on the Main thread."

(and wrap at 80 characters)

Reply:
done